TcgPocket/Features/Users/UsersController.cs

```diff
@@ -0,0 +1,106 @@
+    {
+        var response = await _mediator.Send(new GetUserByIdQuery { Id = id });
+
+        return response.HasErrors ? NotFound(response) : Ok(response);
+    }
+
+    [HttpGet("{id:int}/roles", Name = nameof(GetAllRolesByUserId))]
+    public async Task<ActionResult<Response<List<RoleGetDto>>>> GetAllRolesByUserId([FromRoute] int id)
+    {
+        var response = await _mediator.Send(new GetAllRolesByUserIdQuery { UserId = id });
+
+        return response.HasErrors ? NotFound(response) : Ok(response);
+    }
+
+    [HttpPost]
+    public async Task<ActionResult<Response<UserGetDto>>> CreateUser([FromBody] UserCreateDto data)
+    {
+        var response = await _mediator.Send(new CreateUserCommand { User = data });
+
+        return response.HasErrors
+            ? BadRequest(response)
+            : CreatedAtRoute(nameof(GetUserById), new { response.Data.Id }, response);
+    }
+
+    [HttpPost("{id:int}/roles")]
+    public async Task<ActionResult<Response<UserRoleDto>>> AddRoleToUser([FromRoute] int id, [FromBody] AddRoleToUserCommand data)
+    {
+        var response = await _mediator.Send(new AddRoleToUserCommand { Id = id, RoleName = data.RoleName });
+
+        return response.HasErrors
+            ? BadRequest(response)
+            : CreatedAtRoute(nameof(GetUserById), new { response.Data.Id }, response);
+    }
+
+    [HttpPut("{id:int}")]
+    public async Task<ActionResult<Response<UserGetDto>>> UpdateUser([FromRoute] int id,
+        [FromBody] UserDto data)
+    {
+        var response = await _mediator.Send(new UpdateUserCommand { Id = id, User = data });
+
+        return response.HasErrors ? BadRequest(response) : Ok(response);
+    }
+
+    [HttpPut("{id:int}/password-update", Name = nameof(UpdatePassword))]
+    public async Task<ActionResult<Response<UserGetDto>>> UpdatePassword(
+        [FromBody] UpdatePasswordCommand data)
+    {
+        var response = await _mediator
+            .Send(new UpdatePasswordCommand 
+            { 
+                UserName = data.UserName, 
+                CurrentPassword = data.CurrentPassword, 
+                NewPassword = data.NewPassword, 
+                NewPasswordConfirmation = data.NewPasswordConfirmation 
+            });
+
+        return response.HasErrors ? BadRequest(response) : Ok(response);
+    }
+
+    [HttpDelete("{id:int}")]
+    public async Task<ActionResult<Response>> DeleteUser([FromRoute] int id)
+    {
+        var response = await _mediator.Send(new DeleteUserCommand { Id = id });
+
+        return response.HasErrors ? BadRequest(response) : Ok(response);
+    }
+
+    [HttpDelete("{id:int}/role")]
+    public async Task<ActionResult<Response>> RemoveRoleFromUser([FromRoute] int id, [FromBody] RemoveRoleFromUserCommand data)
+    {
+        var response = await _mediator.Send(new RemoveRoleFromUserCommand { UserId = id, RoleName = data.RoleName });
+
+        return response.HasErrors ? BadRequest(response) : Ok(response);
+    }
```

Comment:
We should just get the `roleName` instead of the entire command.

```suggestion
    [HttpDelete("{id:int}/role")]
    public async Task<ActionResult<Response>> RemoveRoleFromUser([FromRoute] int id, [FromBody] string roleName)
    {
        var response = await _mediator.Send(new RemoveRoleFromUserCommand { UserId = id, RoleName = roleName });

        return response.HasErrors ? BadRequest(response) : Ok(response);
    }
```